File: src/parallel.cpp
#include <nanobind/nanobind.h>
#include <IParallelizable.h>

namespace nb = nanobind;

NB_MODULE(_parallel, m)
{
    nb::enum_<gwm::ParallelType>(m, "_ParallelType")
        .value("SerialOnly", gwm::ParallelType::SerialOnly)
        .value("OpenMP", gwm::ParallelType::OpenMP)
        .value("CUDA", gwm::ParallelType::CUDA)
        ;
}

File: src/pygwmodel/parallel.py
"""Parallel type enumeration for GW models.

``ParallelType`` is an IntEnum with members:

- ``SerialOnly`` — single-threaded execution (default).
- ``OpenMP``   — multi-threaded via OpenMP.
- ``CUDA``     — GPU-accelerated via CUDA.
"""

from ._parallel import _ParallelType

ParallelType = _ParallelType
